CMakeLists.txt

```diff
@@ -18,6 +18,13 @@ option(BUILD_DOCUMENTATION "Create HTML based API documentation (requires Doxyge
 option(RUN_LONG_UNIT_TESTS "Tests that takes more than 5 minutes to complete. No effect if BUILD_TESTING is OFF" OFF)
 option(BUILD_STORAGE_SAMPLES "Build sample application for Azure Storage clients" OFF)
 
+# Use the static runtime libraries when building statically for consistency with vcpkg's
+# arch-windows-static triplets:
+cmake_policy(SET CMP0091 NEW) # https://cmake.org/cmake/help/v3.15/variable/CMAKE_MSVC_RUNTIME_LIBRARY.html
```

Comment:
If this policy isn't available will CMAKE_MSVC_RUNTIME_LIBRARY be respected?  i.e. what happens if we are cmake 3.5 (ignore the fact other stuff in our cmake doesn't work on old cmake versions :) )

Reply:
pre-cmake 3.15 fixes I've found do more harm than good in my opinion. They are ok for the application you know how you are building, but for the SDK library, which you may wish to build with multi-threaded static and dynamic CRT, they are too intrusive, unless we introduce a build flag called something like MSVC_CRT = DYNAMIC | STATIC.
pre-3.15 cmake did not handle this situation well - I think, it is better to show the warning. We used the same approach in C SDK.